File: 1.1/src/cz/mg/vulkan/PFNvkCmdDrawIndexed.java
package cz.mg.vulkan;

public class PFNvkCmdDrawIndexed extends VkFunctionPointer {
    public PFNvkCmdDrawIndexed() {
    }

    protected PFNvkCmdDrawIndexed(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkCmdDrawIndexed(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkCmdDrawIndexed(long value) {
        setValue(value);
    }

    public PFNvkCmdDrawIndexed(VkInstance instance) {
        super(instance, new VkString("vkCmdDrawIndexed"));
    }

    public void call(VkCommandBuffer commandBuffer, int indexCount, int instanceCount, int firstIndex, int vertexOffset, int firstInstance){
        callNative(getValue(), commandBuffer != null ? commandBuffer.getVkAddress() : VkPointer.getNullAddressNative(), indexCount, instanceCount, firstIndex, vertexOffset, firstInstance);
    }

    protected static native void callNative(long vkaddress, long commandBuffer, int indexCount, int instanceCount, int firstIndex, int vertexOffset, int firstInstance);

}

File: 1.1/src/c/cz/mg/vulkan/PFN_vkCmdDrawIndexed.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkCmdDrawIndexed_callNative(JNIEnv* env, jclass jc, jlong address, jlong commandBuffer, jint indexCount, jint instanceCount, jint firstIndex, jint vertexOffset, jint firstInstance){
    (void)env;
    (void)jc;
    PFN_vkCmdDrawIndexed f = (PFN_vkCmdDrawIndexed)jniLongToFunctionPointer(address);
    f(
        *((VkCommandBuffer*)jniLongToPointer(commandBuffer)),
        (uint32_t)indexCount,
        (uint32_t)instanceCount,
        (uint32_t)firstIndex,
        (int32_t)vertexOffset,
        (uint32_t)firstInstance
    );
}

